src/php/aspell.php

```diff
@@ -60,6 +60,12 @@
     // when splitting words.
     $text = preg_replace('/--+/u', ' ', $text);
 
+    // Get rid of e-mail addresses
+    $text = preg_replace("/[^@\s]*@[^@\s]*\.[^@\s]*/", ' ', $text);
+
+    // Get rid of URLs
+    $text = preg_replace("/[a-zA-Z]*[:\/\/]*[A-Za-z0-9\-_]+\.+[A-Za-z0-9\.\/%&=\?\-_]+/i", ' ', $text);
```

Comment:
My gut feeling is that if a word contains a '@' then that probably happened on purpose and it's likely not a word that one would expect/want to be spell checked.  And these days, that includes any type of '@' mention too like (@plobbes) because the at-mention name is often not a normal-ish word either.

So my feeling is that the use case for FILTER_VALIDATE_* is quite different (looking for and finding validity or not) than determining if/when we might want to check the spelling of some word that is URL-ish, email-ish or even @mention-ish.  The less sophisticated we are probably the better.  If special characters like [@:#~] (maybe '/' and possibly others too?) show up (anywhere) in a word then that word is a likely good candidate to ignore.